docs/configuration/http_conn_man/route_config/route.rst

```diff
@@ -338,6 +342,28 @@ header_name
   the request header is not present, the load balancer will use a random number as the hash,
   effectively making the load balancing policy random.
 
+.. _config_http_conn_man_route_table_route_add_req_headers:
+
+Adding custom request headers
+-----------------------------
+
+Custom request headers can be added to a request that matches a specific route. The headers are
+specified in the following form:
+
+.. code-block:: json
+
+  [
+    {"key": "header1", "value": "value1"},
+    {"key": "header2", "value": "value2"}
+  ]
+
+*Note:* In the presence of duplicate header keys, headers specified at the global
```

Comment:
I don't agree with the ordering of precedence. The most specific rules (routes) should take precendence over the less specific rules virtual host and vh over  HTTP Conn Man route configurations.
The order of 
1. Route
2. Virtual Host
3. Route configuration within the HTTP Conn Man Filter. 

I believe this to be better because at the HTTP Conn Man layer we have request headers to be added for all requests then particular vhs or routes can set particular more specific header values.